### PR TITLE
Add Design Principles to the docs

### PR DESCRIPTION
**Pull request by gaearon**
This adds a new “Contributing” subsection to the Docs.
I plan it to be short:

* How to Contribute (basically our Contributing.md)
* Design Principles (this PR!)
* Code Conventions (todo)
* Implementation Overview (todo)

This PR adds Design Principles. I don’t want to sit on this forever so I figured let’s start with this and I’ll add the rest as I write it.

### CODE REVIEW

**Comment by gaearon** [comment]
To give you an idea:

screen shot 2016-07-14 at 20 23 12

(there'll be more)

screen shot 2016-07-14 at 20 23 16

File: docs/contributing/design-principles.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+We don't deprecate anything without a good reason. We recognize that sometimes deprecations warnings cause frustration but we add them because deprecations clean up the road for the improvements and new features that we and many people in the community consider valuable.
+
+For example, we added a [warning about unknown DOM props](https://facebook.github.io/react/warnings/unknown-prop.html) in React 15.2.0. Many projects were affected by this. However fixing this warning is important so that we can introduce the support for [custom attributes](https://github.com/facebook/react/issues/140) to React. There is a reason like this behind every deprecation that we add.
+
+When we add a deprecation warning, we keep it for the rest of the current major version, and [change the behavior in the next major version](/react/blog/2016/02/19/new-versioning-scheme.html). If there is a lot of manual repetitive work involved, we release a [codemod](https://www.youtube.com/watch?v=d0pOgY8__JM) script that automates most of the change. Codemods enable us to move forward without stagnation in a massive codebase, and we encourage you to use them as well.
```
Comment (alexzherdev) [inline docs/contributing/design-principles.md]: Not an expert at all, but I think "repetitive manual work" sounds better than "manual repetitive work".

**Comment by zpao** [comment]
doit

File: docs/contributing/design-principles.md (hunk 1 above)
Comment (gaearon) [inline docs/contributing/design-principles.md]: Fixed, thanks.

**Pull request merged**